packages/client/screens/HomeScreen.tsx

```diff
@@ -106,6 +109,27 @@ const HomeScreen: React.FC<HomeTabHomeScreenScreenProps> = ({ navigation }) => {
                     }}
                 />
 
+                <Button
+                    testID="sign-up-button"
+                    title="signUp"
+                    onPress={async () => {
+                        const { userID, userNickname } = await sendSignUp();
+                        Toast.show({
+                            type: 'success',
+                            text1: 'Signed up successfully',
+                        });
+
+                        localStorage.setItem('USER_ID', userID);
+                        localStorage.setItem(
+                            'USER_CREDENTIALS',
+                            JSON.stringify({
+                                userID,
+                                userNickname,
+                            }),
+                        );
+                    }}
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Async operations not driven by a state machine? What is it?